src/WebClient.spec.js

```diff
@@ -500,15 +501,95 @@ describe('WebClient', function () {
   });
 
   describe('has rate limit handling', function () {
-    it('should expose retry headers in the response');
+    it('should expose retry headers in the response', function() {
+      const scope = nock('https://slack.com')
+        .post(/api/)
+        .reply(200, { ok: true }, {
+          'retry-after': 1
+        });
+      const client = new WebClient(token);
+      return client.apiCall('method')
+        .then((result) => {
+          assert.deepNestedInclude(result, { 'retryAfter': 1000 });
+          scope.done();
+        });
+    });
+
     // NOTE: see retry policy note below
-    it('should allow rate limit triggered retries to be turned off');
+    it('should allow rate limit triggered retries to be turned off', function() {
+      const scope = nock('https://slack.com')
+        .post(/api/)
+        .reply(429, { ok: false }, {
+          'retry-after': 0
+        })
+        .post(/api/)
+        .reply(200, { ok: true });
+      const client = new WebClient(token, { retryConfig: { retries: 0 } });
+      return client.apiCall('method')
+        .then(() => {
+          assert.fail(0, 1, 'Rate Limit retry not turned off');
+        })
+        .catch((error) => {
+          assert.isFalse(scope.isDone());
```

Comment:
it makes sense that you're setting up responses that are never meant to trigger in the scope, and therefore you check `scope.isDone()` rather than calling `scope.done()`. do you think we should also be cleaning up these test cases in a teardown method with `scope.cleanAll()`?